shared/logger.py

```
@@ -35,10 +58,42 @@ def setup_logger(name, level=logging.INFO,
+                # Create a listener that will write to the console
+                console_handler = NonBlockingStreamHandler(sys.stdout)
+                console_handler.setFormatter(logging.Formatter(format, datefmt))
+
+                # Create and start the listener
+                listener = QueueListener(log_queue, console_handler)
+                listener.start()
+
+                # Store the listener reference to prevent garbage collection
+                logger._queue_listener = listener
+
+        except Exception:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)